src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -250,6 +266,87 @@ private static IntPtr GetSslMethod(bool isServer, long options)
             return method;
         }
 
+        private static int VerifyClientCertificate(int preverify_ok, IntPtr x509_ctx_ptr)
+        {
+            using (SafeX509StoreCtxHandle storeHandle = new SafeX509StoreCtxHandle(x509_ctx_ptr, false))
+            {
+                using (var chain = new X509Chain())
+                {
+                    chain.ChainPolicy.RevocationMode = X509RevocationMode.Online;
+                    chain.ChainPolicy.RevocationFlag = X509RevocationFlag.ExcludeRoot;
+
+                    using (SafeX509StackHandle chainStack = libcrypto.X509_STORE_CTX_get1_chain(storeHandle))
+                    {
+                        if (chainStack.IsInvalid)
+                        {
+                            Debug.Fail("Invalid chain stack handle");
+                            return 0;
+                        }
+
+                        IntPtr certPtr = Crypto.GetX509StackField(chainStack, 0);
+                        if (IntPtr.Zero == certPtr)
+                        {
+                            return 0;
+                        }
+
+                        using (X509Certificate2 cert = new X509Certificate2(certPtr))
+                        {
+                            return chain.Build(cert) ? 1 : 0;
+                        }
+                    }
+                }
+            }
+        }
+
+        private static void UpdateCAListFromRootStore(libssl.SafeSslContextHandle context)
+        {
+            using (SafeX509NameStackHandle nameStack = Crypto.NewX509NameStack())
+            {
+                //maintaining the HashSet of Certificate's issuer name to keep track of duplicates 
+                HashSet<string> issuerNameHashSet = new HashSet<string>();
+
+                //Enumerate Certificates from LocalMachine and CurrentUser root store 
+                AddX509Names(nameStack, StoreLocation.LocalMachine, issuerNameHashSet);
+                AddX509Names(nameStack, StoreLocation.CurrentUser, issuerNameHashSet);
+                
+                libssl.SSL_CTX_set_client_CA_list(context, nameStack);
+
+                // The handle ownership has been transferred into the CTX.
+                nameStack.SetHandleAsInvalid();
+            }
+
+        }
+
+        private static void AddX509Names(SafeX509NameStackHandle nameStack, StoreLocation storeLocation, HashSet<string> issuerNameHashSet)
+        {
+            using (var myStore = new X509Store(StoreName.Root, storeLocation))
+            {
+                myStore.Open(OpenFlags.ReadOnly);
+
+                foreach (var certificate in myStore.Certificates)
+                {
+                    //Check if issuer name is already present
+                    //Avoiding duplicate names
+                    if (!issuerNameHashSet.Add(certificate.Issuer)) continue;
```

Comment:
Our style does not permit the if-body to be on the same line as the if.  Please add explicit bracing and newlines.